src/app/shared/store/filter.state.ts

```diff
@@ -244,6 +253,35 @@ export class FilterState {
     patchState({ searchQuery: payload, from: 0 });
   }
 
+  @Action(LoadPreviousResults)
+  loadPreviousResults(ctx: StateContext<FilterStateModel>): void {
+    const previousResults: string[] | undefined = JSON.parse(localStorage.getItem('previousResults')) || [];
+    if (!previousResults?.length) {
+      localStorage.setItem('previousResults', JSON.stringify([]));
+    }
+    ctx.patchState({ previousResults });
+  }
+
+  @Action(AddPreviousResult)
+  addPreviousResult(ctx: StateContext<FilterStateModel>, { result }: AddPreviousResult): void {
+    const state = ctx.getState();
+    const normalizedResult = result.trim().toLowerCase();
+    const normalizedResults = state.previousResults.map((res) => res.trim().toLowerCase());
+
+    if (normalizedResult && !normalizedResults.includes(normalizedResult)) {
+      const updatedResults = [result, ...state.previousResults.slice(0, Constants.MAX_PREVIOUS_SEARCH_RESULTS - 1)];
+      localStorage.setItem('previousResults', JSON.stringify(updatedResults));
```

Comment:
Ideally, we should avoid using localStorage directly and store the value in state.
You can have a look at https://www.ngxs.io/v3.8/plugins/storage

We probably won't be able to store part of the state with another storage engine, but maybe we can move it to meta-data.state or something similar, and then we can move other localStorage usages to that state.